feat: RFD 7 Deployment and Upgrade Benchmark (#1399)
cloudfoundry-incubator/kubecf: doc/rfd/rfd/0007/README.md

---
authors: Jaime Gomes <jaime.gomes@suse.com>
state: draft
discussion: 
---

# RFD 7 Deployment and Upgrade Benchmark

## Introduction

### Motivation

Currently, we do not have any data that can corroborate with some eye evidence that deployment times
of KubeCF are increasing between releases.

### Goal

Be able to measure and collect time metrics during a fresh deployment and between upgrades, on each
component (pod) present on KubeCF.

With the time data collected, both upstream projects and downstream product teams can investigate

> if there're optimizations that can lead to a better and more fasten deployments that will support a

Reviewer: Dislike contraction. Also phrasing.
```
- there're optimizations that can lead to a better and more fasten deployments
+ there are optimizations that can lead to better and faster deployments
```

> if there are optimizations that can lead to a better and more faster deployments that will support a

Reviewer: ```suggestion
if there are optimizations that can lead to better and faster deployments that will support a
```

The `a better` is singular, the `deployments` is plural. Removing the `a` to match that plural.
`faster` itself implies `more`.

better user experience in future releases.

### Metric(s) to collect

| Resource | Metric Pseudo Name              | Condition                 | Namespace |
| -------- | ------------------------------- | ------------------------- | --------- |
| Job      | kube_job_status_completion_time | kube_job_status_succeeded | kubecf    |
| Pod      | kube_pod_completion_time        | kube_pod_status_ready     | kubecf    |

## Possible solution(s)

### Solution A

### Metrics streamer

A k8s native component like [kube-state-metrics[(https://github.com/kubernetes/kube-state-metrics)]

> that is deployable on a pod and it listens the k8s API server and generates metrics. It is

Reviewer: ```suggestion
that is deployable on a pod, monitors the k8s API server, and generates metrics. It is
```

> important that it can be configurable regarding the namespace or namespaces that can collect from

Reviewer: ```suggestion
important that it can be configured regarding the namespace(s) it collects from
```

> and acceptable from a security perspective.

Reviewer: ```suggestion
and is acceptable from a security perspective.
```

### Metrics collector

> Promotheus?

Reviewer: ```suggestion
Prometheus?
```

---
authors: Jaime Gomes <jaime.gomes@suse.com>
state: draft
discussion: 
---

# RFD 7 Deployment and Upgrade Benchmark

## Introduction

### Motivation

Currently, we do not have any gather data that can corroborate with some eye evidence that

> deployment times of KubeCF are increasing and upgrades are not argument supported in the next

Author: typo

> deployment times of KubeCF are increasing and upgrades are not argument supported in the next

Author: @viccuad I think it's _increasing_ but maybe I can check the waters once again

> deployment times of KubeCF are increasing and upgrades are not argument supported in the next

Reviewer 2: Well, I was trying to word it positively :sweat_smile: .

releases.

### Goal

Be able to measure and collect time metrics during a fresh deployment and between upgrades, on each
component present on KubeCF.

With the time data collected, both upstream projects and downstream product teams can investigate
if there're optimizations that can lead to a better and more fasten deployments that will support a
better user experience in future releases.

### Type of Deployments

1. Eirini enabled:
1.1 High availability
2. Diego enabled:
2.1 High availability

> ## Possible solution(s)

Reviewer 2: I would like this RFD to specify the API and kind of data that one receives from the Benchmark, instead of the implementation of it.

Author: must contain both at the ends and I am going to add the "kind of data"
